### PR TITLE
Add OB tests for the engine.

### PR DESCRIPTION
**Pull request by pilhuhn**

### CODE REVIEW

**Comment by pilhuhn** [comment]
This error just makes no sense to me. There is no reason the code should even try to talk to port 80. 
And besides that it works locally 🙄

**Comment by gwenneg** [comment]
ℹ️ I fixed the conflicts.

File: engine/src/test/java/com/redhat/cloud/notifications/MockserverConfig.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+import static org.mockserver.model.RegexBody.regex;
+
+public class MockserverConfig {
+
+    public static void addOpenBridgeEndpoints(Map<String, String> auth, Bridge bridge, Map<String, String> processor) {
+        String authString = Json.encode(auth);
+        String bridgeString = Json.encode(bridge);
+        String processorString = Json.encode(processor);
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/MockserverConfig.java:18]: This is never used.

File: engine/src/test/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessorTest.java [hunk 2]
```diff
@@ -128,6 +136,92 @@ void testCamelEndpointProcessing() {
...
+        // Metrics should report the same thing.
+        micrometerAssertionHelper.assertCounterIncrement(PROCESSED_COUNTER_NAME, 2);
+
+        // Let's have a look at the first result entry fields.
+        historyItem = result.get(0);
+        assertEquals(event, historyItem.getEvent());
+        assertEquals(endpoint, historyItem.getEndpoint());
+        assertEquals(CAMEL, historyItem.getEndpointType());
+        assertEquals("slack", historyItem.getEndpointSubType());
+        assertEquals(null, historyItem.getDetails());
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessorTest.java:199]: Could be `assertNull`.

**Review by gwenneg — APPROVED** [review]
I added two minor comments. Otherwise, LGTM.

File: engine/src/test/java/com/redhat/cloud/notifications/MockserverConfig.java (hunk 1 above)
Comment (gwenneg) [reply engine/src/test/java/com/redhat/cloud/notifications/MockserverConfig.java:18]:
```suggestion
```

File: engine/src/test/java/com/redhat/cloud/notifications/MockserverConfig.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.mockserver.model.ClearType;
+
+import java.util.Map;
+
+import static com.redhat.cloud.notifications.MockServerLifecycleManager.getClient;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+import static org.mockserver.model.RegexBody.regex;
+
+public class MockserverConfig {
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/MockserverConfig.java:13]:
```suggestion
public class MockServerConfig {
```

File: engine/src/test/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessorTest.java [hunk 4]
```diff
@@ -47,6 +51,7 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessorTest.java:54]:
```suggestion
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertTrue;
```

File: engine/src/test/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessorTest.java (hunk 2 above)
Comment (gwenneg) [reply engine/src/test/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessorTest.java:199]:
```suggestion
        assertNull(historyItem.getDetails());
```

**Review by gwenneg — APPROVED** [review]

**Comment by gwenneg** [comment]
Fixing build errors I created 🤦‍♂️

**Comment by gwenneg** [comment]
Fixing build errors I created 🤦‍♂️

**Comment by pilhuhn** [comment]
Thanks @gwenneg for taking care!

**Pull request merged**